### PR TITLE
EPMDJ-5237: Support a new api for creating a list of agents

### PR DESCRIPTION
**Pull request by RomanAchilov**
Issue: EPMDJ-5237 EPMDJ-5005

### CODE REVIEW

File: src/pages/agents-page/agent-status-toggler/agent-status-toggler.tsx [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
+  className?: string;
+  agent: Agent;
+  onChange: () => void;
+}
+
+export const AgentStatusToggler = ({ className, agent, onChange }: Props) => (
+  <div className={className}>
+    <Panel data-test="agent-status-toggler">
+      {agent.status === AGENT_STATUS.BUSY ? (
+        <Inputs.Toggler value={agent.status === AGENT_STATUS.BUSY} label={<Spinner />} disabled />
```
Comment (RomanDavlyatshin) [inline src/pages/agents-page/agent-status-toggler/agent-status-toggler.tsx:17]: 1. `value={agent.status === AGENT_STATUS.BUSY}` contains a redundand comparison.
2. Why use ternary operator just to render disabled `<Inputs.Toggler />`? I get that we have to calculate `label` and `disabled` values, but we can do that outside of JSX.
3. I'm pretty sure that such a component is called `Toggle` not a `Toggler` 😃

File: src/pages/agents-page/agents-page.tsx [hunk 2]
```diff
@@ -21,24 +21,18 @@ const agentsPage = BEM(styles);
 
 export const AgentsPage = agentsPage(({ className }: Props) => {
   const { push } = useHistory();
-  const { single = [], grouped = [] } = useWsConnection<{ single: Agent[]; grouped: ServiceGroup[] }>(
-    defaultAdminSocket,
-    '/agents',
-  ) || {};
   const agentsList = useWsConnection<Agent[]>(
     defaultAdminSocket,
```
Comment (RomanDavlyatshin) [inline src/pages/agents-page/agents-page.tsx:25]: A hook that requires passing a socket instance adds a meaningless burden to any component that uses it.
Can't hook obtain socket on its own?

File: src/pages/agents-page/agent-status-toggler/agent-status-toggler.tsx [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
+import * as React from 'react';
+import { Panel, Spinner, Inputs } from '@drill4j/ui-kit';
+
+import { AGENT_STATUS } from 'common/constants';
+import { Agent } from 'types/agent';
+
+interface Props {
+  className?: string;
+  agent: Agent;
+  onChange: () => void;
```
Comment (RomanDavlyatshin) [inline src/pages/agents-page/agent-status-toggler/agent-status-toggler.tsx:10]: It is not `<AgentStatusToggle />` if it cannot toggle status on its own.
Why passing `onChange`?  `<AgentStatusToggle />` can call `api/toggleAgent` method directly

File: src/pages/agents-page/agents-table/agents-table.tsx [hunk 4]
```diff
@@ -85,7 +85,7 @@ export const AgentsTable = agentsTable(({ className, agents }: Props) => {
           Cell={({ value, item }) => {
             const isOfflineAgent = item.agentType === 'Java' && !item.agentVersion;
             return (item.agentType !== 'ServiceGroup' && !isOfflineAgent ? (
-              <AgentStatusToggler status={value} onChange={() => toggleAgent(item.id, value, showMessage)} />
+              <AgentStatusToggler agent={item} onChange={() => toggleAgent(item.id, value, showMessage)} />
```
Comment (RomanDavlyatshin) [inline src/pages/agents-page/agents-table/agents-table.tsx:88]: `toggleAgent` should not be aware of `showMessage`
In that concrete case just do something like that:
```
onChange={async () => {
  try {
    await toggleAgent(item.id, value)
    showMessage('success')
  } catch (e) {
    showMessage('failure: ${e.message}')
  }
}}
```
But as I've said, I would much prefer if AgentStatusToggle were self-sufficient, managing API calls internally.
It also might manage success/failure messages internally, but that depends on notifications implementation.

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: src/pages/agents-page/agents-page.tsx (hunk 2 above)
Comment (RomanAchilov) [reply src/pages/agents-page/agents-page.tsx:25]: This hook can receive one of two sockets. And is used until topic+the socket won't start repeating when this happens creating a more specialized hook

**Review by RomanDavlyatshin — APPROVED** [review]

**Pull request merged**